File: apps/android/app/src/main/java/chat/mural/network/AIProvider.kt
package chat.mural.network

import android.content.Context

enum class AIProvider(
    val displayName: String,
    val helperModel: String,
    val liveModel: String,
    val keyUrl: String,
    val usageUrl: String,
    val dataControlsUrl: String,
) {
    OPENAI("OpenAI", "gpt-5.6-luna", "gpt-live-1", "https://platform.openai.com/api-keys", "https://platform.openai.com/usage", "https://developers.openai.com/api/docs/guides/your-data"),
    GOOGLE_AI_STUDIO("Google AI Studio", "gemma-4-31b-it", "gemini-3.8-live", "https://aistudio.google.com/app/apikey", "https://aistudio.google.com/app/usage", "https://ai.google.dev/gemini-api/docs/usage-policies"),
}

object AIProviderSelection {
    private const val PREFERENCES = "mural_ai_provider"
    private const val PROVIDER = "provider"

    fun read(context: Context): AIProvider {
        val raw = context.applicationContext.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE)
            .getString(PROVIDER, null)
        return AIProvider.entries.firstOrNull { it.name == raw } ?: AIProvider.OPENAI
    }

    fun save(context: Context, provider: AIProvider) {
        context.applicationContext.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE)
            .edit().putString(PROVIDER, provider.name).apply()
    }
}

File: apps/ios/App/AIProvider.swift
import Foundation

enum AIProvider: String, CaseIterable, Identifiable {
    case openAI
    case googleAIStudio

    var id: String { rawValue }

    var title: String {
        switch self {
        case .openAI: "OpenAI"
        case .googleAIStudio: "Google AI Studio"
        }
    }

    var helperModel: String {
        switch self {
        case .openAI: "gpt-5.6-luna"
        case .googleAIStudio: "gemma-4-31b-it"
        }
    }

    var liveModel: String {
        switch self {
        case .openAI: "gpt-live-1"
        case .googleAIStudio: "gemini-3.8-live"
        }
    }

    var keyURL: URL {
        switch self {
        case .openAI: URL(string: "https://platform.openai.com/api-keys")!
        case .googleAIStudio: URL(string: "https://aistudio.google.com/app/apikey")!
        }
    }

    var usageURL: URL {
        switch self {
        case .openAI: URL(string: "https://platform.openai.com/usage")!
        case .googleAIStudio: URL(string: "https://aistudio.google.com/app/usage")!
        }
    }

    var dataControlsURL: URL {
        switch self {
        case .openAI: URL(string: "https://developers.openai.com/api/docs/guides/your-data")!
        case .googleAIStudio: URL(string: "https://ai.google.dev/gemini-api/docs/usage-policies")!
        }
    }

    var keyPlaceholder: String {
        switch self {
        case .openAI: "OpenAI API key"
        case .googleAIStudio: "Google AI Studio API key"
        }
    }

    var keyHelp: String {
        switch self {
        case .openAI: "Stored only on this iPhone. Mural sends it directly to OpenAI."
        case .googleAIStudio: "Stored only on this iPhone. Mural sends it directly to Google AI Studio."
        }
    }
}

enum AIProviderSelection {
    private static let defaultsKey = "mural.ai-provider"

    static var current: AIProvider {
        guard let rawValue = UserDefaults.standard.string(forKey: defaultsKey),
              let provider = AIProvider(rawValue: rawValue) else { return .openAI }
        return provider
    }

    static func save(_ provider: AIProvider) {
        UserDefaults.standard.set(provider.rawValue, forKey: defaultsKey)
    }
}
